stan/math/rev/core/var.hpp

```diff
@@ -181,6 +184,56 @@ namespace stan {
       // NOLINTNEXTLINE
       var(unsigned long x) : vi_(new vari(static_cast<double>(x))) { }  // NOLINT
 
+      /**
+       * Construct a variable from the specified arithmetic argument
+       * by constructing a new <code>vari</code> with the argument
+       * cast to <code>double</code>, and a zero adjoint. Only works
+       * if the imaginary part is zero.
+       *
+       * @param x Value of the variable.
+       */
+      explicit var(std::complex<double> x) {
+        assert(imag(x) == 0);
+        vi_ = new vari(real(x));
+      }
+
+      /**
+       * Construct a variable from the specified arithmetic argument
+       * by constructing a new <code>vari</code> with the argument
+       * cast to <code>double</code>, and a zero adjoint. Only works
+       * if the imaginary part is zero.
+       *
+       * @param x Value of the variable.
+       */
+      explicit var(std::complex<float> x) {
+        assert(imag(x) == 0);
+        vi_ = new vari(static_cast<double>(real(x)));
+      }
+
+      /**
+       * Construct a variable from the specified arithmetic argument
+       * by constructing a new <code>vari</code> with the argument
+       * cast to <code>double</code>, and a zero adjoint. Only works
+       * if the imaginary part is zero.
+       *
+       * @param x Value of the variable.
+       */
+      explicit var(std::complex<long double> x) {
+        assert(imag(x) == 0);
+        vi_ = new vari(static_cast<double>(real(x)));
+      }
+
+      /**
+       * This constructor does not work and really only exists to
+       * ensure that such a var is not created accidentally in
+       * testing.
+       *
+       * @param x Value of the variable.
+       */
+      explicit var(std::complex<var> x) {
+        assert(false);
```

Comment:
I don't think you need this constructor at all.  If you try to call `var(complex<var>(2,0))` it won't have a constructor to match, so the compiler will catch it.  If `complex<T>` can be cast to `T` (I'm not sure, but I'd be surprised), then you want to declare this constructor but not define it.  That will prevent it from being instantiated at compile time.

```
explicit var(const std::complex<var>& x);
```

Reply:
This does not prevent the segfault in #123 but neither does filling in the body to only throw an exception. I don't know what to do but am pushing anyway.

Reply:
The segfault is from `complex<var>(var(1))` not `var(complex)`.   

I don't think we can specialize the `complex<T>` constructor to `complex<var>` to fail.